packages/@uppy/core/src/index.js

```diff
@@ -553,16 +553,14 @@ class Uppy {
 
     this.setState({
       currentUploads: updatedUploads,
-      files: updatedFiles
+      files: updatedFiles,
+      ...(
+        // If this is the last file we just removed - allow new uploads!
+        Object.keys(updatedFiles).length === 0 &&
+        { allowNewUpload: true }
+      )
```

Comment:
Is `...(` necessary because you don’t want to `allowNewUpload: false` if this was not the last file?

I mean:

```js
const isLastFile = Object.keys(updatedFiles).length === 0
setState({
  allowNewUpload: isLastFile
})
```

Reply:
@arturi, it is necessary, exactly, tests won't pass if we set it to `false` here. 
Logic is very stateful, we can't judge that 'upload is always allowed if `this.state.files === 0`'. This line states that 'if we removeFile() the last file, and we haven't uploaded anything yet (or cancelled all of it), - allow uploads again', and nothing more.